### PR TITLE
Redis caching for query responses

### PR DESCRIPTION
**Pull request by andreivasilescu24**

### CODE REVIEW

File: microservice/src/routes/helpers.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+use chrono::DateTime;
+use multiversx_sc_snippets::imports::RustBigUint;
+
+pub fn denominate(value: f64) -> u128 {
```
Comment (mihaicalinluca) [inline microservice/src/routes/helpers.rs:4]: small unit tests for these functions would be nice

File: microservice/src/routes/setup.rs [hunk 2]
```diff
@@ -43,5 +52,29 @@ pub async fn setup_contract(
             new_address_bech32.clone(),
         ));
 
+    let mut con = redis_client
+        .get_multiplexed_async_connection()
+        .await
+        .unwrap();
+
+    // Invalidate values corresponding to previous deployed contract
+    let _: () = con.del("user_addresses").await.unwrap();
```
Comment (mihaicalinluca) [inline microservice/src/routes/setup.rs]: is there a way to get rid of all this data in a single call?

File: microservice/src/routes/tx_models.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+impl DeployResponse {
+    pub fn new(tx_response: (String, String)) -> Self {
+        Self {
+            response: tx_response.0,
+            address: tx_response.1,
+        }
+    }
+
+    pub fn send(&self) -> HttpResponse {
```
Comment (mihaicalinluca) [inline microservice/src/routes/tx_models.rs]: rename this into something like "response" (as previously), or "json", something that can suggest the data is only transformed and no action is actually taken with the request

File: microservice/src/routes/tx_models.rs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+pub struct PingResponse {
+    response: String,
+}
+
+impl PingResponse {
+    pub fn new(response: String) -> Self {
+        Self { response }
+    }
+
+    pub fn send(&self) -> HttpResponse {
```
Comment (mihaicalinluca) [inline microservice/src/routes/tx_models.rs]: same as before

File: microservice/src/routes/tx_models.rs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+pub struct PongResponse {
+    response: String,
+}
+
+impl PongResponse {
+    pub fn new(response: String) -> Self {
+        Self { response }
+    }
+
+    pub fn send(&self) -> HttpResponse {
```
Comment (mihaicalinluca) [inline microservice/src/routes/tx_models.rs]: same as before

**Review by mihaicalinluca — CHANGES_REQUESTED** [review]

File: microservice/src/routes/helpers.rs (hunk 1 above)
Comment (andreivasilescu24) [reply microservice/src/routes/helpers.rs:4]: `denominate` function unit tests coming in PR #3 because it was refactored

**Review by mihaicalinluca — APPROVED** [review]

**Pull request merged**